data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/resistance/resistance_to_damage_type_special.cfg

```diff
@@ -0,0 +1,266 @@
+            [/effect]
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    [resistance]
+                        id=arcane_reduce
+                        affect_self=yes
+                        apply_to=arcane,fire
+                        value=50
+                        max_value=50
+                        active_on=defense
+                    [/resistance]
+                [/abilities]
+            [/effect]
+            [effect]
+                apply_to=attack
+                [set_specials]
+                    mode=append
+                    [attacks]
+                        value=1
+                    [/attacks]
+                    [damage]
+                        value=12
+                    [/damage]
+                [/set_specials]
+            [/effect]
+            [filter]
+                id=bob
+            [/filter]
+        [/object]
+        [object]
+            silent=yes
+            [effect]
+                apply_to=attack
+                [set_specials]
+                    mode=append
+                    [attacks]
+                        value=1
+                    [/attacks]
+                    [damage]
+                        value=12
+                    [/damage]
+                    [damage_type]
+                        alternative_type=arcane
+                    [/damage_type]
+                    [chance_to_hit]
+                        value=100
+                    [/chance_to_hit]
+                [/set_specials]
+            [/effect]
+            [filter]
+                id=alice
+            [/filter]
+        [/object]
+
+        [store_unit]
+            [filter]
+                id=alice
+            [/filter]
+            variable=a
+            kill=yes
+        [/store_unit]
+        [store_unit]
+            [filter]
+                id=bob
+            [/filter]
+            variable=b
+        [/store_unit]
+        [unstore_unit]
+            variable=a
+            find_vacant=yes
+            x,y=$b.x,$b.y
+        [/unstore_unit]
+        [store_unit]
+            [filter]
+                id=alice
+            [/filter]
+            variable=a
+        [/store_unit]
+
+        [do_command]
+            [attack]
+                weapon=0
+                defender_weapon=0
+                [source]
+                    x,y=$a.x,$a.y
+                [/source]
+                [destination]
+                    x,y=$b.x,$b.y
+                [/destination]
+            [/attack]
+        [/do_command]
+        [store_unit]
+            [filter]
+                id=bob
+            [/filter]
+            variable=a
+        [/store_unit]
+        #damage without modification are 12, if test fail hitpoints !=82
+        #if succed then damage by alice 18(bob more vulnerable to arcane but resistance ability can't detect damage_type)
+        {ASSERT ({VARIABLE_CONDITIONAL a.hitpoints equals 82})}
```

Comment:
Similar to https://github.com/wesnoth/wesnoth/pull/8639#discussion_r1546728908 , please use the COMMON_KEEP_... and ATTACK_AND_VALIDATE macros to make the test code more readable.